src/term-spec.js

```diff
@@ -128,21 +128,133 @@ declare export class AssignmentTargetPropertyProperty extends AssignmentTargetPr
 // class
 declare export class ClassExpression extends Expression {
   name?: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassDeclaration extends Statement {
+  isAbstract: any; // boolean
   name: BindingIdentifier;
-  super?: Expression;
-  elements: ClassElement[];
+  typeParameters?: TypeParameterDeclaration[];
+  heritageClauses?: HeritageClause[];
+  elements: (ClassElement | IndexSignatureDeclaration)[];
 }
 
 declare export class ClassElement extends Term {
-  isStatic: any;
+  isStatic: any; // boolean
+  accessModifier: any; // 'public' | 'protected' | 'private'
+}
+
+declare export class MethodClassElement extends ClassElement {
+  isAbstract: any; // boolean
+  hasQuestionToken: any; // boolean
   method: MethodDefinition;
 }
 
+declare export class ConstructorClassElement extends ClassElement {
+  params: ConstructorFormalParameters;
+  body: FunctionBody;
+}
+
+declare export class ConstructorFormalParameters extends Term {
+  items: ConstructorParameterDeclaration[];
+  rest?: ParameterDeclaration;
+}
+
+declare export class ConstructorParameterDeclaration extends Term {
+  accessModifier?: any; // 'public' | 'protected' | 'private'
+  hasReadonlyModifier: any; // boolean
+  parameter: ParameterDeclaration;
+}
```

Comment:
@disnet Now that I'm looking at it again, it might be nicer to have `ConstructorParameterDeclaration` inherit from `ParameterDeclaration`.